AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1327 @@
+			return;
+	}
+
+	if (az_result_failed(result = parse_twin_desired_property(payload_span, &twin_properties)))
+	{
+		// If the item can't be found, the desired temp might not be set so take no action
+		debug_printError("  MAIN: Could not parse desired property, return code %d\n", result);
+	}
+	else
+	{
+		update_leds(&twin_properties);
+		send_reported_property(&twin_properties);
+	}
+}
 
-/*
-    Main application
- */
+/**********************************************
+* Callback for device twin
+**********************************************/
+void receivedFromCloud_property(uint8_t* topic, uint8_t* payload)
+{
+	debug_printInfo("  MAIN: Device Property Get");
+
+	if (topic == NULL)
+	{
+		debug_printWarn("  MAIN: NULL topic");
+		return;
+	}
+
+	az_iot_pnp_client_property_response property_response;
+	az_span property_topic = az_span_create_from_str((char *)topic);
+	az_result rc = az_iot_pnp_client_property_parse_received_topic(&pnp_client, property_topic, &property_response);
+	RETURN_WITH_MESSAGE_IF_FAILED(rc, "az_iot_pnp_client_property_parse_received_topic failed");
+
+	if (az_span_size(property_response.request_id) != 0 && IOT_DEBUG_PRINT)
+	{
+		char request_id_buf[50];
+		az_span_to_str(request_id_buf, sizeof(request_id_buf), property_response.request_id);
+		debug_printInfo("  MAIN: Property request, request_id:%s, status: %d", request_id_buf, property_response.status);
+	}
+	if (payload == NULL)
+	{
+		debug_printWarn("  MAIN: NULL payload");
+	}
+	else
+	{
+		handle_property_message(az_span_create_from_str((char*)payload), &property_response);
+	}
+}
 
+/**********************************************
+* Callback for device twin patch
+**********************************************/
+void receivedFromCloud_patch(uint8_t* topic, uint8_t* payload)
+{
+	debug_printInfo("  MAIN: Device Property Patch");
+
+	if (topic == NULL)
+	{
+		debug_printWarn("  MAIN: NULL topic");
+		return;
+	}
+
+	az_iot_pnp_client_property_response property_response;
+	az_span property_topic = az_span_create_from_str((char *)topic);
+	az_result rc = az_iot_pnp_client_property_parse_received_topic(&pnp_client, property_topic, &property_response);
+	RETURN_WITH_MESSAGE_IF_FAILED(rc, "az_iot_pnp_client_property_parse_received_topic failed");
+
+	if (payload == NULL)
+	{
+		debug_printWarn("  MAIN: NULL payload");
+	}
+	else
+	{
+		handle_property_message(az_span_create_from_str((char*)payload), &property_response);
+	}
+}
+
+/**********************************************
+* Check if button(s) was pressed.
+**********************************************/
+void check_button_status(void)
+{
+	az_span button_event_span = AZ_SPAN_FROM_BUFFER(button_event_payload);
+	az_json_writer jw;
+
+	// save flags in case user pressed buttons very fast, and for error case.
+	bool sw0 = button_press_data.flag.sw0 == 1 ? true : false;
+	bool sw1 = button_press_data.flag.sw1 == 1 ? true : false;
```

Comment:
booleans are often easier to use with verbs attached to the name
```suggestion
	bool sw0_pressed = button_press_data.flag.sw0 == 1 ? true : false;
	bool sw1_pressed = button_press_data.flag.sw1 == 1 ? true : false;
```

leads to below suggestions

Reply:
Done
[commit](https://github.com/daisukeiot/Microchip-PIC-IoT-Wx-1/commit/78a33ee2f255993b7b48e1c06e1b697ec03caf30)